src/robotFactory.js

```diff
@@ -1,10 +1,75 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.chipVersion = chipVersion;
 
-class FlyingRobot {}
+    if (coords.x === undefined) {
+      this.coords = {
+        x: 0,
+        y: 0,
+      };
+    } else {
+      this.coords = coords;
+    }
+  }
 
-class DeliveryDrone {}
+  goForward(step = 1) {
+    this.coords.y += step;
+  };
+
+  goBack(step = 1) {
+    this.coords.y -= step;
+  };
+
+  goLeft(step = 1) {
+    this.coords.x -= step;
+  };
+
+  goRight(step = 1) {
+    this.coords.x += step;
+  };
+  getInfo() {
+    return `Robot: ${this.name}, Chip version: ${this.chipVersion},`
+    + ` Weight: ${this.weight}`;
+  }
+}
+
+class FlyingRobot extends BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    super(name, weight, coords, chipVersion);
+    this.coords.z = this.coords.z || 0;
+  }
+
+  goUp(step = 1) {
+    this.coords.z += step;
+  }
+
+  goDown(step = 1) {
+    this.coords.z -= step;
+  }
+}
+
+class DeliveryDrone extends FlyingRobot {
+  constructor(name, weight, coords, chipVersion,
+    maxLoadWeight, currentLoad = null) {
```

Comment:
each property on a new line
```suggestion
  constructor(
  	name,
  	weight,
  	coords,
  	chipVersion,
    maxLoadWeight,
    currentLoad = null,
  ) {
```